packages/itwinui-react/src/core/Tile/TileNew.tsx

```diff
@@ -0,0 +1,646 @@
+
+type TileThumbnailPictureOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.ThumbnailAvatar component
+
+const TileThumbnailAvatar = React.forwardRef((props, ref) => {
+  const { as: Element = 'div', className, children, ...rest } = props;
+  if (React.isValidElement(children)) {
+    return React.createElement(children.type, {
+      ...children.props,
+      className: cx('iui-thumbnail-icon', children.props.className),
+    });
+  }
+
+  return (
+    <Element className={cx(className)} ref={ref} {...rest}>
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'div', TileThumbnailAvatarOwnProps>;
+
+TileThumbnailPicture.displayName = 'TileNew.TileThumbnailAvatar';
+
+type TileThumbnailAvatarOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.QuickAction component
+
+const TileQuickAction = React.forwardRef((props, ref) => {
+  const { as: Element = 'button', className, children, ...rest } = props;
+  return (
+    //todo: styling is all off, need to fix after
+    <Element
+      className={cx('iui-button', 'iui-tile-thumbnail-quick-action', className)}
+      data-iui-size='small'
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'button', TileQuickActionOwnProps>;
+TileQuickAction.displayName = 'Tile.QuickAction';
+
+type TileQuickActionOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.TypeIndicator component
+
+const TileTypeIndicator = React.forwardRef((props, ref) => {
+  const { as: Element = 'button', className, children, ...rest } = props;
+  return (
+    //todo: styling is all off, need to fix after
+    <Element
+      className={cx(
+        'iui-button',
+        'iui-tile-thumbnail-type-indicator',
+        className,
+      )}
+      data-iui-size='small'
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'button', TileTypeIndicatorOwnProps>;
+TileTypeIndicator.displayName = 'Tile.TypeIndicator';
+
+type TileTypeIndicatorOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.Badge component
+
+const TileBadge = polymorphic.div('iui-tile-thumbnail-badge-container');
+TileBadge.displayName = 'Tile.Badge';
+
+type TileBadgeOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.Name component
+
+const TileName = React.forwardRef(
+  ({ as: Element = 'div', children, ...rest }, ref) => {
```

Comment:
I'd prefer if `props` were extracted within the function like all of the other components. Just replace this list with `props` and then extract these in the next line

Reply:
Yea that makes sense. I updated this.